xla/service/gpu/gpu_hlo_cost_analysis.h

```diff
@@ -39,6 +39,9 @@ class GpuHloCostAnalysis : public HloCostAnalysis {
 
   Status Preprocess(const HloInstruction* hlo) override;
 
+  float ScalingRatio(const HloInstruction& hlo) const;
+  int64_t NumOfDevices(const HloInstruction& hlo) const;
```

Comment:
unused?

Reply:
this is actually used in the performance model to determine number of nccl channels.